src/main/java/org/wso2/carbon/module/swiftiso20022/utils/MT940ValidationUtils.java

```diff
@@ -0,0 +1,469 @@
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateClosingAvailableBalance((String) fields
+                .get(ConnectorConstants.CLOSING_AVAIL_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+
+        errorModel = validateForwardAvailableBalance((String) fields
+                .get(ConnectorConstants.FORWARD_CLOSING_AVAIL_BALANCE));
+        if (errorModel.isError()) {
+            return errorModel;
+        }
+        return new ErrorModel();
+    }
+
+    /**
+     * Method to validate whether Transaction reference is valid.
+     *
+     * @param transactionRef  Transaction reference to be validated
+     * @return    Whether transaction reference is valid
+     */
+    public static ErrorModel validateTransactionReference(String transactionRef) {
+        String[] referenceDetails = transactionRef.split(ConnectorConstants.COLON);
+        String reference = null;
+
+        if (referenceDetails.length == 3) {
+            reference = referenceDetails[2];
+        } else {
+            return new ErrorModel(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_INVALID,
+                            ConnectorConstants.TRANSACTION_REF));
+        }
+
+        if (StringUtils.isBlank(reference)) {
+            return new ErrorModel(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_MISSING,
+                            ConnectorConstants.TRANSACTION_REF));
+        }
+
+        if (reference.length() > 16) {
+            return new ErrorModel(ConnectorConstants.ERROR_M50,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_LENGTH,
+                            ConnectorConstants.TRANSACTION_REF, 16));
+        }
+
+        if (reference.startsWith(ConnectorConstants.SLASH) || reference.endsWith(ConnectorConstants.SLASH) ||
+                reference.contains(ConnectorConstants.DOUBLE_SLASH)) {
+            return new ErrorModel(ConnectorConstants.ERROR_T26,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_CONTAINS_SLASH,
+                            ConnectorConstants.TRANSACTION_REF));
+        }
+
+        if (!Pattern.matches(ConnectorConstants.MT_REGEX_PATTERN, reference)) {
+            return new ErrorModel(ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_INVALID,
+                            ConnectorConstants.TRANSACTION_REF));
+        }
+        return new ErrorModel();
+    }
+
+    /**
+     * Method to validate whether Related reference is valid.
+     *
+     * @param relatedRef  Related reference to be validated
+     * @return    Whether related reference is valid
+     */
+    public static ErrorModel validateRelatedReference(String relatedRef) {
```

Comment:
This seems almost similar to `validateTransactionReference` method and few other methods. Can't we move the common logic to a single place?

Reply:
FIxed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/befc8f745ba4f7a2cb0b949ac349d98c62e8ec5b